packages/storefront-api-client/src/storefront-api-client.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  createGraphQLClient,
+  CustomFetchAPI,
+  RequestParams as GQLClientRequestParams,
+} from "@shopify/graphql-client";
+
+import {
+  StorefrontAPIClient,
+  SFAPIClientConfig,
+  SFAPIClientRequestOptions,
+} from "./types";
+import {
+  DEFAULT_SDK_VARIANT,
+  DEFAULT_CLIENT_VERSION,
+  SDK_VARIANT_HEADER,
+  SDK_VARIANT_SOURCE_HEADER,
+  SDK_VERSION_HEADER,
+  DEFAULT_CONTENT_TYPE,
+  PUBLIC_ACCESS_TOKEN_HEADER,
+  PRIVATE_ACCESS_TOKEN_HEADER,
+} from "./constants";
+import {
+  getCurrentSupportedAPIVersions,
+  validateRequiredStoreDomain,
+  validateRequiredAccessTokens,
+  validatePrivateAccessTokenUsage,
+  validateApiVersion,
+} from "./utilities";
+
+const httpRegEx = new RegExp("^(https?:)?//");
+
+export function createStorefrontAPIClient({
+  storeDomain,
+  apiVersion,
+  publicAccessToken,
+  privateAccessToken,
+  clientName,
+  customFetchAPI: clientFetchAPI,
+}: {
+  storeDomain: string;
+  apiVersion: string;
+  clientName?: string;
+  customFetchAPI?: CustomFetchAPI;
+} & (
+  | {
+      publicAccessToken?: never;
+      privateAccessToken: string;
+    }
+  | {
+      publicAccessToken: string;
+      privateAccessToken?: never;
+    }
+)): StorefrontAPIClient {
+  const currentSupportedAPIVersions = getCurrentSupportedAPIVersions();
+
+  validateRequiredStoreDomain(storeDomain);
+  validateApiVersion(currentSupportedAPIVersions, apiVersion);
+  validateRequiredAccessTokens(publicAccessToken, privateAccessToken);
+  validatePrivateAccessTokenUsage(privateAccessToken);
+
+  const trimmedStoreDomain = storeDomain.trim();
```

Comment:
Thoughts on leaning on the URL API here? We could skip the trim, cleanup and `endsWith` check further down.

```js
const { host: domain } = new URL("https://test.myshopify.com/ ")
console.log(domain) // => test.myshopify.com
```

Reply:
Good call out! ✅